examples/xgboost_cv.py

```diff
@@ -0,0 +1,109 @@
+        print("please create folder to store xgboost CV results")
+
+# Set Constants
+seed = 108
+n_folds = 3
+
+# Make the Optuna Sampler behave in a deterministically.
+sampler = TPESampler(seed=seed)
+
+
+# Define Optuna Objective
+# FYI: Objective functions can take additional arguments
+# (https://optuna.readthedocs.io/en/stable/faq.html#objective-func-additional-args).
+def objective(trial):
+    (data, target) = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    dtrain = xgb.DMatrix(data, label=target)
+
+    param = {
+        "silent": 1,
+        "objective": "binary:logistic",
+        "eval_metric": "auc",
+        "booster": trial.suggest_categorical("booster", ["gbtree"]),
+        "alpha": trial.suggest_loguniform("alpha", 1e-3, 1.0),
+    }
+
+    if param["booster"] == "gbtree" or param["booster"] == "dart":
+        param["max_depth"] = trial.suggest_int("max_depth", 1, 9)
+        param["min_child_weight"] = trial.suggest_int("min_child_weight", 1, 9)
+        param["eta"] = trial.suggest_loguniform("eta", 1e-3, 1.0)
+        param["gamma"] = trial.suggest_loguniform("gamma", 1e-3, 1.0)
+        param["subsample"] = trial.suggest_loguniform("subsample", 0.6, 1.0)
+        param["colsample_bytree"] = trial.suggest_loguniform("colsample_bytree", 0.6, 1.0)
+        param["grow_policy"] = trial.suggest_categorical("grow_policy", ["depthwise", "lossguide"])
+
+    if param["booster"] == "dart":
+        param["sample_type"] = trial.suggest_categorical("sample_type", ["uniform", "weighted"])
+        param["normalize_type"] = trial.suggest_categorical("normalize_type", ["tree", "forest"])
+        param["rate_drop"] = trial.suggest_loguniform("rate_drop", 1e-8, 1.0)
+        param["skip_drop"] = trial.suggest_loguniform("skip_drop", 1e-8, 1.0)
+
+    xgb_cv_results = xgb.cv(
+        params=param,
+        dtrain=dtrain,
+        num_boost_round=10000,
+        nfold=n_folds,
+        stratified=True,
+        early_stopping_rounds=100,
+        seed=seed,
+        verbose_eval=False,
+    )
+
+    # (Optional)
+    # Print n_estimators in the output at each call to the objective function
+    print(
+        "-" * 10,
+        "Trial {} has optimal trees: {}".format(trial.number, str(xgb_cv_results.shape[0])),
+        "-" * 10,
+    )
+    print()
+
+    # (Optional: Disable if not needed)
+    # Save XGB results for Analysis; Update to your path by changing: file_path
+    xgb_cv_results.to_csv(file_path + "Optuna_cv_{}.csv".format(trial.number), index=False)
+
+    # Set n_estimators as a trial attribute; Accessible via study.trials_dataframe()
+    trial.set_user_attr("n_estimators", len(xgb_cv_results))
+
+    # Extract the best score
+    best_score = xgb_cv_results["test-auc-mean"].values[-1]
+    return best_score
+
+
+if __name__ == "__main__":
+    study = optuna.create_study(direction="maximize", sampler=sampler)
+    study.optimize(objective, n_trials=10)
+    print()
+    print(study.best_trial)
```

Comment:
Some examples show the best trial, but I think it's kind for users to show the formatted output because this example has many hyperparameters.

Example output:
```console
FrozenTrial(number=7, value=0.991965, datetime_start=datetime.datetime(2020, 5, 15, 7, 17, 45, 351583), datetime_complete=datetime.datetime(2020, 5, 15, 7, 17, 45, 657804), params={'booster': 'gbtree', 'alpha': 0.005432114697045233, 'max_depth': 6, 'min_child_weight': 4, 'eta': 0.25584322384906694, 'gamma': 0.01993575229218288, 'subsample': 0.7217599909772354, 'colsample_bytree': 0.8655936942560539, 'grow_policy': 'lossguide'}, distributions={'booster': CategoricalDistribution(choices=('gbtree',)), 'alpha': LogUniformDistribution(high=1.0, low=0.001), 'max_depth': IntUniformDistribution(high=9, low=1, step=1), 'min_child_weight': IntUniformDistribution(high=9, low=1, step=1), 'eta': LogUniformDistribution(high=1.0, low=0.001), 'gamma': LogUniformDistribution(high=1.0, low=0.001), 'subsample': LogUniformDistribution(high=1.0, low=0.6), 'colsample_bytree': LogUniformDistribution(high=1.0, low=0.6), 'grow_policy': CategoricalDistribution(choices=('depthwise', 'lossguide'))}, user_attrs={'n_estimators': 57}, system_attrs={}, intermediate_values={}, trial_id=7, state=TrialState.COMPLETE)
```

I suggest using a format similar to [examples/pruning/xgboost_cv_integration.py](https://github.com/optuna/optuna/blob/master/examples/pruning/xgboost_cv_integration.py).
```suggestion
    print("Best trial:")
    trial = study.best_trial

    print("  Value: {}".format(trial.value))

    print("  Params: ")
    for key, value in trial.params.items():
        print("    {}: {}".format(key, value))

    print("  Number of estimators: {}".format(trial.user_attrs["n_estimators"]))
```

Reply:
Changed